ocrd/ocrd/resolver.py

```diff
@@ -100,7 +105,46 @@ def download_to_directory(self, directory, url, basename=None, if_exists='skip',
             dst_path.write_bytes(src_path.read_bytes())
         else:
             log.debug("Downloading URL '%s' to '%s'", url, dst_path)
-            response = requests.get(url)
+            if 'OCRD_DOWNLOAD_RETRIES' in environ:
+                retries = retries or int(environ['OCRD_DOWNLOAD_RETRIES'])
+            retries = retries or 0
+            for _ in range(retries + 1):
+                try:
+                    if timeout is None and 'OCRD_DOWNLOAD_TIMEOUT' in environ:
+                        timeout = environ['OCRD_DOWNLOAD_TIMEOUT'].split(',')
+                        if len(timeout) > 1:
+                            timeout = tuple(float(x) for x in timeout)
+                        else:
+                            timeout = float(timeout[0])
+                    response = requests.get(url, timeout=timeout)
+                    if response.status_code in [
+                            # probably too wide:
+                            408, # Request Timeout
+                            409, # Conflict
+                            410, # Gone
```

Comment:
If not 404 then also not 410.

Reply:
Indeed – agreed!
```suggestion
```

Reply:
I'll just wait for the CI before committing.